cmd/security-agent/subcommands/start/command.go

```diff
@@ -123,6 +128,58 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 					}
 					return tagger.NewTaggerParams()
 				}),
+				fx.Provide(func() startstop.Stopper {
+					return startstop.NewSerialStopper()
+				}),
+				fx.Provide(func(config config.Component, statsd statsd.Component) (ddgostatsd.ClientInterface, error) {
+					return statsd.CreateForHostPort(pkgconfig.GetBindHost(), config.GetInt("dogstatsd_port"))
+				}),
+				fx.Provide(func(stopper startstop.Stopper, log log.Component, config config.Component, statsdClient ddgostatsd.ClientInterface, demultiplexer demultiplexer.Component) (status.InformationProvider, error) {
+					hostnameDetected, err := utils.GetHostnameWithContextAndFallback(context.TODO())
+					if err != nil {
+						return status.NewInformationProvider(nil), err
+					}
+
+					runtimeAgent, err := runtime.StartRuntimeSecurity(log, config, hostnameDetected, stopper, statsdClient, demultiplexer)
+					if err != nil {
+						return status.NewInformationProvider(nil), err
+					}
+
+					if runtimeAgent == nil {
+						return status.NewInformationProvider(nil), nil
+					}
+
+					return status.NewInformationProvider(runtimeAgent.StatusProvider()), nil
+				}),
+				fx.Provide(func(stopper startstop.Stopper, log log.Component, config config.Component, statsdClient ddgostatsd.ClientInterface, demultiplexer demultiplexer.Component, sysprobeconfig sysprobeconfig.Component) (status.InformationProvider, error) {
+					hostnameDetected, err := utils.GetHostnameWithContextAndFallback(context.TODO())
+					if err != nil {
+						return status.NewInformationProvider(nil), err
+					}
+
+					// start compliance security agent
+					complianceAgent, err := compliance.StartCompliance(log, config, sysprobeconfig, hostnameDetected, stopper, statsdClient, demultiplexer)
+					if err != nil {
+						return status.NewInformationProvider(nil), err
+					}
+
+					if complianceAgent == nil {
+						return status.NewInformationProvider(nil), nil
+					}
+
+					return status.NewInformationProvider(complianceAgent.StatusProvider()), nil
```

Comment:
Same here about `runtimeAgent` and `complianceAgent`. Shouldn't we keep a ref somewhere on this to avoid the GC cleaning those ?

Reply:
Both runtimeAgent and complianceAgent ref are stored in the different status providers:

-  https://github.com/DataDog/datadog-agent/blob/bc2e4776002657fca462fb25533551ce3415ec3f/pkg/security/agent/status_provider.go#L24-L28
- https://github.com/DataDog/datadog-agent/blob/bc2e4776002657fca462fb25533551ce3415ec3f/pkg/compliance/status_provider.go#L25-L29

Then we hold to the status providers in the status component. 

Would that be enough to avoid GC clean those references?

Reply:
Done https://github.com/DataDog/datadog-agent/pull/23263/commits/0850d2496b54669647de0d7e74dd3d06fe383522